cai_causal_graph/graph_components.py

```diff
@@ -292,24 +317,109 @@ def variable_name(self) -> str:
-        if not super().__eq__(other):
+        if not super().__eq__(other, deep):
             return False
         return self.variable_name == other.variable_name and self.time_lag == other.time_lag
 
     def __hash__(self) -> int:
         """Return a hash value of the node identifier."""
         return super().__hash__()
 
+    def to_dict(self, include_meta: bool = True) -> dict:
+        """
+        Serialize a `cai_causal_graph.graph_components.TimeSeriesNode` instance to a dictionary.
+
+        :param include_meta: Whether to include meta information about the node in the dictionary. Default is `True`.
+        :return: The dictionary representation of the `cai_causal_graph.graph_components.TimeSeriesNode` instance.
+        """
+        dictionary = super().to_dict(include_meta)
+        # add the time lag and variable name to the dictionary
+        dictionary.update({TIME_LAG: self.time_lag, VARIABLE_NAME: self.variable_name})
+        return dictionary
+
+    @classmethod
+    def from_dict(cls, dictionary: dict) -> TimeSeriesNode:
+        """
+        Return a `cai_causal_graph.graph_components.TimeSeriesNode` instance from a dictionary.
+
+        :param dictionary: The dictionary from which to create the time series node.
+        :return: The time series node created from the dictionary.
+        """
+        assert 'identifier' in dictionary, 'The dictionary must contain the `identifier` key.'
+
+        # check the meta for the time lag and variable name match the ones in the dictionary
+        time_lag = dictionary.get(TIME_LAG, None)
+        variable_name = dictionary.get(VARIABLE_NAME, None)
+
+        if 'meta' not in dictionary:
+            dictionary['meta'] = {}
+
+        if time_lag is None:
+            time_lag = dictionary['meta'].get(TIME_LAG, None)
+        else:
+            # check that the time lag in the meta matches the time lag in the dictionary
+            time_lag_meta = dictionary['meta'].get(TIME_LAG, None)
+
+            if time_lag != time_lag_meta:
+                dictionary['meta'][TIME_LAG] = time_lag
+                logger.warning(
+                    f'The time lag in the meta ({time_lag_meta}) does not match the time lag in the dictionary ({time_lag}).'
+                )
+
+        if variable_name is None:
+            variable_name = dictionary['meta'].get(VARIABLE_NAME, None)
+        else:
+            # check that the variable name in the meta matches the variable name in the dictionary
+            variable_name_meta = dictionary['meta'].get(VARIABLE_NAME, None)
+
+            if variable_name != variable_name_meta:
+                dictionary['meta'][VARIABLE_NAME] = variable_name
+                logger.warning(
+                    f'The variable name in the meta ({variable_name_meta}) does not match the variable name in the '
+                    f'dictionary ({variable_name}).'
```

Comment:
Would say you are setting meta one to the dictionary one, i.e. overwriting the current meta value.